common/src/main/java/com/microsoft/identity/common/internal/telemetry/adapter/TelemetryAggregationAdapter.java

```diff
@@ -106,4 +102,45 @@ public void process(@NonNull final List<Map<String, String>> rawData) {
 
         return nonPiiProperties;
     }
+
+    //The response time is the duration of the last occurrence.
+    private void trackEventResponseTime(@NonNull final Map<String, String> responseTimeMap,
+                                        @NonNull final Map<String, String> event) {
+        final String eventName = event.get(Key.EVENT_NAME);
+        final String eventType = event.get(Key.EVENT_TYPE);
+
+        if (eventName.contains(START)) {
+            final String eventStartTime = eventType + "_start_time";
+            responseTimeMap.put(
+                    eventStartTime,
+                    null == event.get(Key.OCCUR_TIME) ?
+                            null
+                            : event.get(Key.OCCUR_TIME)
+            );
+        }
+
+        if (eventName.contains(END)) {
+            final String eventEndTime = eventType + "_end_time";
+            responseTimeMap.put(
+                    eventEndTime,
+                    null == event.get(Key.OCCUR_TIME) ?
+                            null
+                            : event.get(Key.OCCUR_TIME)
+            );
+        }
+    }
+
+    private void calculateEventResponseTime(@NonNull final Map<String, String> responseTimeMap,
+                                            @NonNull final Map<String, String> aggregatedData) {
+        for (Map.Entry<String,String> entry : responseTimeMap.entrySet()) {
+            if (entry.getKey().contains(START)
+                    && responseTimeMap.containsKey(entry.getKey().replace(START, END))
```

Comment:
This `if` is the really challenging part to read here - isn't this check equivalent to....

```java
if (entry.getKey().contains(START) &&
    responseTimeMap.containsKey(END) &&
    responseTimeMap.get(END) != null) {
        // ...
}
```